File: vite-project/src/types/medicine.ts
import type { PageResult, Result } from './common';

// 药品分类DTO
export interface MedicineCategoryDTO {
  id?: number;
  name: string;
  description?: string;
  sort?: number;
  status?: number;
}

// 药品DTO
export interface MedicineDTO {
  id?: number;
  name: string;
  images?: string;
  manufacturer?: string;
  specification?: string;
  price?: number;
  stock?: number;
  description?: string;
  usageMethod?: string;
  contraindication?: string;
  status?: number;
  categoryId: number;
  categoryName?: string;
}

// API 响应类型
export type MedicineListResult = Result<PageResult<MedicineDTO>>;
export type MedicineDetailResult = Result<MedicineDTO>;
export type MedicineCategoryListResult = Result<MedicineCategoryDTO[]>;

// 查询参数类型
export interface MedicineQueryParams {
  keyword?: string;
  categoryId?: number;
  page?: number;
  size?: number;
} 

File: backend/src/main/java/com/example/backend/modules/medicine/entity/Medicine.java
package com.example.backend.modules.medicine.entity;

import com.example.backend.common.base.BaseEntity;
import com.example.backend.modules.medicine.enums.MedicineStatusEnum;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import java.math.BigDecimal;

@Data
@Entity
@Table(name = "t_medicine")
@EqualsAndHashCode(callSuper = true)
public class Medicine extends BaseEntity {

    @Column(nullable = false, length = 100)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private MedicineCategory category;

    @Column(length = 100)
    private String specification;

    @Column(length = 200)
    private String manufacturer;

    @Column(precision = 10, scale = 2)
    private BigDecimal price;

    private Integer stock;

    @Column(columnDefinition = "text")
    private String description;

    @Column(columnDefinition = "text")
    private String usageMethod;

    @Column(columnDefinition = "text")
    private String contraindication;

    @Column(length = 1000)
    private String images;

    @Enumerated(EnumType.ORDINAL)
    private MedicineStatusEnum status = MedicineStatusEnum.NORMAL;
} 